src/poetry/core/pyproject/toml.py

```diff
@@ -28,8 +28,15 @@ def data(self) -> dict[str, Any]:
             if not self.path.exists():
                 self._data = {}
             else:
-                with self.path.open("rb") as f:
-                    self._data = tomllib.load(f)
+                try:
+                    with self.path.open("rb") as f:
+                        self._data = tomllib.load(f)
+                except tomllib.TOMLDecodeError as e:
+                    from poetry.core.pyproject.exceptions import PyProjectException
+
+                    raise PyProjectException(
+                        f"poetry cannot parse {self._path.as_posix()}"
```

Comment:
I do not like the wording. Mentioning "poetry" does not give any useful information (and strictly speaking it is not completely correct because we are in `poetry-core`, which can be used as a standalone build backend).

I'd propose either just `"Could not parse ..."`, which is used in other places or maybe better `"... is not a valid TOML file"` or similar so that it is clear that not just Poetry could not parse the toml file but it is just invalid in general.

> Should we try to include the toml parse error at 0 verbosity somehow?

I think so, for example `"... is not a valid TOML file: {e}"`

Reply:
Yes, it does. Doing `poetry run python big_project.py`  it's very helpful to know it's not that complex 1000 files web app, but instead the package manager failing.

Reply:
>I think so, for example "... is not a valid TOML file: {e}"

Ahh cool, that's pretty elegant.

I do think there's value in mentioning that `poetry` (or `poetry-core`) is what has encountered the error, because of the case @luckydonald mentioned. You won't necessarily know which tool to pass `-v` too. (`poetry run -v python big_project.py` vs `poetry run python big_project.py -v`) But if if all the information to see what the problem is is there are 0 verbosity then this is less of a big deal.

Here's an example of another error message that mentions poetry: https://github.com/python-poetry/poetry-core/blob/94f933f50ade83cb7f13a2c521477aa4b8a0b1ee/src/poetry/core/factory.py#L494